Bump version to 0.0.1a7 (#516)
astral-sh/ty: CHANGELOG.md

# Changelog

## 0.0.1-alpha.7

### Bug fixes

- Implement Python's floor division semantics for `Literal` `int`s ([#18249](https://github.com/astral-sh/ruff/pull/18249))

> - Don't warn `yield` not in function when `yield` is in function ([#18008](https://github.com/astral-sh/ruff/pull/18008))

Reviewer: ```suggestion
- Don't warn about a `yield` expression not being in a function if the `yield` expression is in a function ([#18008](https://github.com/astral-sh/ruff/pull/18008))
```

> - Fix attribute writes to unions/intersections including modules ([#18313](https://github.com/astral-sh/ruff/pull/18313))

Reviewer: ```suggestion
- Fix inference of attribute writes to unions/intersections that including module-literal types ([#18313](https://github.com/astral-sh/ruff/pull/18313))
```

> - Fix binary intersection comparison inference logic ([#18266](https://github.com/astral-sh/ruff/pull/18266))

Reviewer: ```suggestion
- Fix false-positive diagnostics in binary comparison inference logic for intersection types ([#18266](https://github.com/astral-sh/ruff/pull/18266))
```

- Fix instance vs callable subtyping/assignability ([#18260](https://github.com/astral-sh/ruff/pull/18260))
- Ignore `ClassVar` declarations when resolving instance members ([#18241](https://github.com/astral-sh/ruff/pull/18241))

> - Infer types for `ty_extensions.Intersection[A, B]` tuple expressions ([#18321](https://github.com/astral-sh/ruff/pull/18321))

Reviewer: (Rephrase it to make it slightly more tailored to the change a user is likely to be interested in)

```suggestion
- Fix crash when hovering over a `ty_extensions.Intersection[A, B]` expression in an IDE context ([#18321](https://github.com/astral-sh/ruff/pull/18321))
```

> - Respect `MRO_NO_OBJECT_FALLBACK` policy when looking up symbols on `type` instances ([#18312](https://github.com/astral-sh/ruff/pull/18312))

Reviewer: sort of an internal change?

```suggestion
```

Author: I mean, it's not the most impactful change in the world, but it's not internal. It even has some ecosystem changes.

- `get_protocol_members` returns a frozenset, not a tuple ([#18284](https://github.com/astral-sh/ruff/pull/18284))

### Typing semantics and features

- Support `import <namespace>` and `from <namespace> import module` ([#18137](https://github.com/astral-sh/ruff/pull/18137))
- Support frozen dataclasses ([#17974](https://github.com/astral-sh/ruff/pull/17974))

> - Use `__getattribute__` to lookup unknown members on a type ([#18280](https://github.com/astral-sh/ruff/pull/18280))

Reviewer: ```suggestion
- Understand that the presence of a `__getattribute__` method indicates arbitrary members can exist on a type ([#18280](https://github.com/astral-sh/ruff/pull/18280))
```

- Add a subdiagnostic if `invalid-return-type` is emitted on a method with an empty body on a non-protocol subclass of a protocol class ([#18243](https://github.com/astral-sh/ruff/pull/18243))

> - Do not carry the generic context of `Protocol` or `Generic` in the `ClassBase` enum ([#17989](https://github.com/astral-sh/ruff/pull/17989))

Reviewer: seems kinda internal

```suggestion
```

- Improve `invalid-type-form` diagnostic where a module-literal type is used in a type expression and the module has a member which would be valid in a type expression ([#18244](https://github.com/astral-sh/ruff/pull/18244))
- Split `invalid-base` error code into two error codes ([#18245](https://github.com/astral-sh/ruff/pull/18245))
- Rename `call-possibly-unbound-method` to `possibly-unbound-implicit-call` ([#18017](https://github.com/astral-sh/ruff/pull/18017))

### Configuration

> - Add `tests` to `src.root` if it exists and is not a package ([#18286](https://github.com/astral-sh/ruff/pull/18286))

Reviewer: ```suggestion
- Add `tests` to `src.root` by default if a `tests/` directory exists and is not a package ([#18286](https://github.com/astral-sh/ruff/pull/18286))
```

- Tell the user why we inferred the Python version we inferred ([#18082](https://github.com/astral-sh/ruff/pull/18082))

> - Resolving Python path using `CONDA_PREFIX` variable to support Conda and Pixi ([#18267](https://github.com/astral-sh/ruff/pull/18267))

Reviewer: ```suggestion
- Add support for detecting activated Conda and Pixi environments ([#18267](https://github.com/astral-sh/ruff/pull/18267))
```

> - Move `respect-ignore-files` under `src` section ([#18322](https://github.com/astral-sh/ruff/pull/18322))

Reviewer: ```suggestion
- Move `respect-ignore-files` configuration setting under `src` section ([#18322](https://github.com/astral-sh/ruff/pull/18322))
```

### Server

- Fix server panic when calling `system_mut` ([#18252](https://github.com/astral-sh/ruff/pull/18252))
- Abort process if worker thread panics ([#18211](https://github.com/astral-sh/ruff/pull/18211))
- Gracefully handle salsa cancellations and panics in background request handlers ([#18254](https://github.com/astral-sh/ruff/pull/18254))

### Playground

- Better default settings for the playground ([#18316](https://github.com/astral-sh/ruff/pull/18316))

> - Use git-commit as ty playground version instead of 0.0.0 ([#18314](https://github.com/astral-sh/ruff/pull/18314))

Reviewer 2: I would omit the playground as it isn't part of the release

### Documentation

- Add long help for `--config` argument ([#18285](https://github.com/astral-sh/ruff/pull/18285))
- Add setup guide for using language server in Neovim ([#436](https://github.com/astral-sh/ty/pull/436))

> - Add warning that docs are autogenerated ([#18270](https://github.com/astral-sh/ruff/pull/18270))

Reviewer: ```suggestion
```

- Clarify `ty check` output default in documentation. ([#18246](https://github.com/astral-sh/ruff/pull/18246))
- Document sources of autogenerated reference docs in contributor guide ([#490](https://github.com/astral-sh/ty/pull/490))

> - Fix anchor tag in `README` ToC ([#486](https://github.com/astral-sh/ty/pull/486))

Reviewer: ```suggestion
```

- Add type compendium ([#18263](https://github.com/astral-sh/ruff/pull/18263))

> - Add ToC and troubleshooting guidance to detailed README ([#464](https://github.com/astral-sh/ty/pull/464))

Reviewer 2: Nit: We omit documentation changes most of the time for ruff (they're a bit noisy)

Author: Removed.
